hostplumber/controllers/hostconfig_controller.go

```diff
@@ -83,12 +84,103 @@ func (r *HostNetworkTemplateReconciler) Reconcile(req ctrl.Request) (ctrl.Result
 		log.Info("SriovConfig is empty")
 	}
 
+	// Everything that is traditonally done under "ifconfig <ifname>" handled here
+	// Alternatively newer "ip addr" and "ip link" - see https://www.redhat.com/sysadmin/ifconfig-vs-ip
+	// MTUs, IPs, routes, link up/down, etc...
+	ifConfigList := hostConfigReq.Spec.InterfaceConfig
+	if len(ifConfigList) > 0 {
+		if err := applyInterfaceConfig(ifConfigList); err != nil {
+			log.Error(err, "Failed to apply interfaceConfig")
+			return ctrl.Result{}, err
+		}
+	} else {
+		log.Info("interfaceConfig is empty")
+	}
+
 	hni := hoststate.New(r.NodeName, r.Namespace, r.Client)
 	hni.DiscoverHostState()
 
 	return ctrl.Result{}, nil
 }
 
+func applyInterfaceConfig(ifConfigList []plumberv1.InterfaceConfig) error {
+	for _, ifConfig := range ifConfigList {
+		var ifName string
+
+		// TODO: nil check needed? This does NOT have an omitempty json tag in _types.go
+		ifName = *ifConfig.Name
+
+		if ifConfig.MTU != nil && *ifConfig.MTU >= 576 {
+			log.Info("interfaceConfig MTU", "MTU", *ifConfig.MTU)
+			if err := linkutils.SetMtuForPf(ifName, *ifConfig.MTU); err != nil {
+				log.Error(err, "Failed to set MTU for ifName", "ifName", ifName, "MTU", *ifConfig.MTU)
```

Comment:
Without deep knowledge of this service, it would probably be more difficult to guess what `ifName` means than what `interface` refers to. (I think)
```suggestion
				log.Error(err, "Failed to set MTU for interface", "interface", ifName, "MTU", *ifConfig.MTU)
```